### ISSUE TITLE
Join command fails to resolve nested references if used in multiple files

### ISSUE THREAD

**Issue by MatthiasGabriel** [Redocly/redocly-cli#1268]
**Describe the bug**

redocly join seems to have issues with resolving nested references if the object itself is referenced in multiple files.
If only the object is referenced only in one file the resolving works perfectly fine.

**To Reproduce**

I uploaded a showcase to https://github.com/MatthiasGabriel/redocly-join-showcase

The following command fails
`redocly join --without-x-tag-groups .\api.yaml .\foo\get\Get.yaml .\foo\create\Create.yaml .\foo\update\Update.yaml .\foo\list\List.yaml`

`[1] foo\shared\NestedFoo.yaml:4:5 at #/properties/bar

Can't resolve $ref

2 | properties:
3 |   bar:
4 |     $ref: './Bar.yaml'
5 |   fooBar:
6 |     type: string

Error was generated by the bundler rule.

❌ Errors encountered while bundling C:\Users\Matthias\Repositories\raiffeisen\devlan2\redocly-test\foo\create\Create.yaml: join will not proceed.
`

**Expected behavior**

**Logs**

-

**OpenAPI description**

-

**Redocly Version(s)**

1.2.0

**`Node.js` Version(s)**

v20.4.0

**Additional context**

All "single" endpoints itself get rendered correctly.
E.g. `redocly join --without-x-tag-groups .\api.yaml .\foo\get\Get.yaml`

If NestedFoo is replaced by SimpleFoo everything renders correctly.

Our original problem appeared when using subtypes (SubFoo), however I think the NestedFoo shows the problem a bit simpler.

**Comment by tatomyr** [comment]
Thanks for reporting the bug!

**Comment by unional** [comment]
I think I have a similar issue, which in my case involves referencing "sibling" references:

```yaml
# api-a/openapi.yaml
paths:
  /somepath:
    $ref: paths/somepath.yaml

# api-a/paths/somepath.yaml
get:
  ...snip...
  responses:
    "400":
      $ref: ../../common-spec/component/responses/problem.400.yaml

# common-spec/component/responses/problem.400.yaml
...snip...
content:
  application/problem+json:
    schema:
      # fail with "Can't resolve $ref here"
      $ref: ../schemas/problem.400.yaml
```

**Comment by IgorKarpiuk** [comment]
Should be fixed in `1.2.1` version

**Comment by MatthiasGabriel** [comment]
Hi @tatomyr  @IgorKarpiuk 

I can confirm that version 1.2.1 fixes my issue. 
Thank you very much for you effort :)
